packages/@webex/plugin-meetings/src/locus-info/mediaSharesUtils.ts

```diff
@@ -188,6 +189,21 @@ MediaSharesUtils.getShareInstanceId = (mediaShares: object) => {
   return extractContent.floor.shareInstanceId;
 };
 
+/**
+ * get deviceUrl that is requesting the floor for media shares (content)
+ * @param {Object} mediaShares
+ * @returns {Object}
+ */
+MediaSharesUtils.getDeviceUrlRequestingShare = (mediaShares: object) => {
+  const contentFloor = MediaSharesUtils.extractContentFloor(mediaShares);
+
+  if (!contentFloor || !contentFloor.requester || !contentFloor.requester.deviceUrl) {
+    return null;
+  }
+
+  return contentFloor.requester.deviceUrl;
```

Comment:
I think we need to use the `beneficiary.deviceUrl` instead of the `requester.deviceUrl`. In most cases these 2 will be the same, but theoretically they can be different and we don't care who asked for share floor, but who actually has it. So, we would need to change this line and line 200 and rename the function to `getContentBeneficiaryDeviceUrl`